lib/core/Observer.ts

```diff
@@ -69,6 +78,22 @@ export default class Observer {
     this.continuePeriodicProcessing = false;
   }
 
+  private async pullLatestBlockChainTime () {
+    try {
+      const latestBlockchainTime = await this.blockchain.getLatestTime();
+      const serviceState = await this.serviceStateStore.get();
+      serviceState.approximateTime = latestBlockchainTime.time;
+      await this.serviceStateStore.put(serviceState);
+      EventEmitter.emit(EventCode.SidetreeBlockchainTimeChanged, { time: latestBlockchainTime.time });
+    } catch (e) {
+      Logger.error(`Error occured while updating blockchain time, investigate and fix: ${e}`);
+    } finally {
+      if (this.continuePeriodicProcessing) {
+        setTimeout(async () => this.pullLatestBlockChainTime(), this.blockchainTimePullIntervalInSeconds * 1000);
+      }
+    }
+  }
+
```

Comment:
Optional design idea: what if this logic also goes into the `BlockchainClock` class since there is already a timer there? Then we decide whether to trigger time update based on a switch we passed in to the constructor? This way we remove the odd dependency that request handling has with Observer (through the service state store), and abstract it with `BlockchainClock`.